components/num-util/src/uint_iterator.rs

```diff
@@ -0,0 +1,59 @@
+use num_integer::Integer;
+use num_traits::{FromPrimitive, ToPrimitive, Unsigned};
```

Comment:
could we skip it for now and just support primitive types?

Reply:
I'm not opposed to that.  However, we should do so knowing the functionality we lose:

- No support for `num::BigUint`.
- No support for arbitrary-precision numbers (limited by an `isize`), unless we want to consider other ways of achieving this, such as `FromStr`.  (I would rather leave`FromStr` to a later ticket, as it is somewhat involved.)